api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -301,6 +301,46 @@ message HttpConnectionManager {
     type.http.v3.PathTransformation http_filter_transformation = 2;
   }
 
+  // Configures the manner in which the Proxy-Status HTTP response header is
+  // populated.
+  //
+  // See the Proxy-Status RFC: https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-proxy-status-05
+  // TODO(ambuc): Update this with the non-draft URL when finalized.
+  //
+  // The Proxy-Status header is a string of the form:
+  //
+  //   "<server_name>; error=<error_type>; details=<details>"
+  message ProxyStatusConfig {
+    enum ProxyName {
+      // Use the string "envoy".
+      ENVOY_LITERAL = 0;
+
+      // Use the name of the server as it appears in
+      // HttpConnectionManager.server_name.
+      SERVER_NAME = 1;
+    }
+
+    // If true, attaches the Proxy-Status header to response headers.
+    // This value defaults to `false`, i.e. no Proxy-Status header is attached.
+    bool attach_proxy_status = 1;
+
+    // If true, populates the details field of the Proxy-Status header with
+    // stream_info.response_code_details.
+    // This value defaults to `false`, i.e. no `details` field is attached.
+    bool attach_details = 2;
+
+    // If true, overwrites the existing Status header with the response code
+    // recommended by the Proxy-Status spec.
+    // This value defaults to `false`, i.e. the HTTP response code is not
+    // overwritten.
+    bool set_recommended_response_code = 3;
```

Comment:
Are the proxy status codes different than the spec? If not I'd be inclined to not have this in config but runtime-guard move Envoy towards HTTP standards recomendations.

Reply:
I think implies that on internal errors Envoy sends response status that is different from what the draft Proxy-Status spec recommends, right? If so do we know what the differences are?